tracer/tracer-opentracing/src/main/java/com/alipay/sofa/rpc/tracer/sofatracer/RpcSofaTracer.java

```diff
@@ -372,6 +379,62 @@ public void clientReceived(SofaRequest request, SofaResponse response, Throwable
         }
     }
 
+    private String generateClientSpecialTimeMarkSpan(RpcInternalContext context) {
+        Long sendTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_CLIENT_SEND_TIME_MICRO);
+        Long receiverTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_CLIENT_RECEIVE_TIME_MICRO);
+        StringBuilder sb = new StringBuilder(calculateMicroTime(sendTime).toString());
+        return sb.append("&").append(calculateMicroTime(receiverTime).toString()).toString();
+    }
+
+    private String generateClientPhaseTimeCostSpan(RpcInternalContext context){
+
+        TreeMap<String, String> resultMap = new TreeMap<>();
+        Long routerTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_CLIENT_ROUTER_TIME_NANO);
+        Long connTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_CONN_CREATE_TIME_NANO);
+        Long filterTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_CLIENT_FILTER_TIME_NANO);
+        Long balancerTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_CLIENT_BALANCER_TIME_NANO);
+        Long reqSerializeTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_REQ_SERIALIZE_TIME_NANO);
+        Long respDeSerializeTime = (Long) context.getAttachment(RpcConstants.INTERNAL_KEY_RESP_DESERIALIZE_TIME_NANO);
+        resultMap.put(TracerRecord.R1.toString(), calculateNanoTime(routerTime).toString());
+        resultMap.put(TracerRecord.R2.toString(), calculateNanoTime(connTime).toString());
+        resultMap.put(TracerRecord.R3.toString(), calculateNanoTime(filterTime).toString());
+        resultMap.put(TracerRecord.R4.toString(), calculateNanoTime(balancerTime).toString());
+        resultMap.put(TracerRecord.R5.toString(),appendResult(calculateNanoTime(reqSerializeTime).toString(), calculateNanoTime(respDeSerializeTime).toString()));
+
+        StringBuilder sb = new StringBuilder();
+
+        for (Map.Entry<String, String> entry : resultMap.entrySet()) {
+            sb.append(entry.getKey());
+            sb.append("=");
+            sb.append(entry.getValue());
+            sb.append("&");
+        }
+        return sb.toString();
+    }
+
+    private Long calculateNanoTime(Long time) {
+        if (time == null) {
+            return -1L;
+        }
+        return TimeUnit.MICROSECONDS.convert(time, TimeUnit.NANOSECONDS);
+    }
+
+    private Long calculateMicroTime(Long time) {
```

Comment:
method do noting ?

Reply:
check time null and return -1,not return time